src/internal/group_api/mod.rs

```diff
@@ -351,6 +384,63 @@ pub(crate) fn get_group_keys<'a>(
     )
 }
 
+fn compare_users<T: Eq + std::hash::Hash + std::fmt::Debug, X>(
+    desired_users: &Vec<T>,
+    found_users: HashMap<T, X>,
+) -> IronOxideErr {
+    let desired_users_set: HashSet<&T> = HashSet::from_iter(desired_users);
+    let found_users_vec: Vec<T> = found_users.into_iter().map(|(x, _)| x).collect();
+    let found_users_set: HashSet<&T> = HashSet::from_iter(&found_users_vec);
+    let diff: Vec<&&T> = desired_users_set.difference(&found_users_set).collect();
+    IronOxideErr::UserDoesNotExist(format!("Failed to find the following users: {:?}", diff))
+}
+
+fn collect_admin_and_member_info<CR: rand::CryptoRng + rand::RngCore>(
+    recrypt: &Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
+    signing_key: &crate::internal::DeviceSigningKeyPair,
+    group_priv_key: recrypt::api::PrivateKey,
+    admins: Vec<UserId>,
+    members: Vec<UserId>,
+    user_ids_and_keys: HashMap<UserId, PublicKey>,
+) -> Result<
+    (
+        Vec<(UserId, PublicKey, TransformKey)>,
+        Vec<(UserId, PublicKey)>,
+    ),
+    IronOxideErr,
+> {
+    let members_set: HashSet<_> = HashSet::from_iter(&members);
+    let admins_set: HashSet<_> = HashSet::from_iter(&admins);
+    let mut member_info: Vec<Result<(UserId, PublicKey, TransformKey), IronOxideErr>> = vec![];
+    let mut admin_info: Vec<(UserId, PublicKey)> = vec![];
+    user_ids_and_keys
+        .into_iter()
+        .for_each(|(id, user_pub_key)| {
+            if members_set.contains(&id) {
+                let maybe_transform_key = recrypt.generate_transform_key(
+                    &group_priv_key.clone().into(),
+                    &user_pub_key.clone().into(),
+                    &signing_key.into(),
+                );
+                match maybe_transform_key {
+                    Ok(member_trans_key) => member_info.push(Ok((
+                        id.clone(),
+                        user_pub_key.clone(),
+                        member_trans_key.into(),
+                    ))),
+                    Err(_) => (),
```

Comment:
We should probably do something with this error case.

Reply:
Good catch. I had meant to push that error into the `member_info` vector so that the `.collect()` at the end would successfully fail on the `?`. I did that now.